lib/vmpooler/api/v1.rb

```diff
@@ -509,6 +514,47 @@ def need_token!
 
       JSON.pretty_generate(result)
     end
+
+    post "#{api_prefix}/vm/:hostname/snapshot/?" do
+      content_type :json
+
+      status 404
+      result = { 'ok' => false }
+
+      params[:hostname] = hostname_shorten(params[:hostname], config['domain'])
+
+      if backend.exists('vmpooler__vm__' + params[:hostname])
+        result[params[:hostname]] = {}
+
+        o = [('a'..'z'), ('0'..'9')].map(&:to_a).flatten
+        result[params[:hostname]]['snapshot'] = o[rand(25)] + (0...31).map { o[rand(o.length)] }.join
```

Comment:
I would ticket the creation of the method and work on it separately. It is out-of-scope of this PR, IMO.